src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,61 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename,
+                             std::ios_base::iostate flag,
+                             std::ios_base::openmode mode)
+    : filename_(filename), flag_(flag), mode_(mode)
+{
+  if (fs::exists(filename_)) {
+    fs::remove(filename_);
+  }
+  std::string tmp_filename = filename_ + ".tmp";
+  if (fs::exists(tmp_filename)) {
+    fs::remove(tmp_filename);
+  }
+  os_.exceptions(flag | std::ofstream::failbit);
+  os_.open(tmp_filename, mode);
```

Comment:
The exception thrown here does not include a filename, that would be nice to have in the exception. Articulate a an exception with filename when file can not be created.

Reply:
What does the exception contain?  Its odd as the filename is a arg.

Reply:
I was going by memory. Could very well be wrong, which would be good!